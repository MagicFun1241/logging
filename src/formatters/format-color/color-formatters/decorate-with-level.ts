import {getDuration, getPackageAndFilename} from '../prefix';
import {indentedBox, indentAllExceptFirstLine, wrap, nonBreakingWhitespace} from '../helpers';
import {getColumns} from '../../../config';

const formatInfo = (prefix: string, content: string) => {
    const columns = getColumns();
    const wrappedContent = wrap(`${prefix.replace(/\s/g, nonBreakingWhitespace)}${nonBreakingWhitespace}${content}`, columns);
    return indentAllExceptFirstLine(wrappedContent);
};

const formatWarn = (prefix: string, content: string) => {
    const text = indentedBox(`WARNING\n\n${content}`, {padding: 1, borderColor: 'yellow'});
    return `${prefix}\n${text}`;
};

const formatError = (prefix: string, content: string) => {
    const text = indentedBox(`ERROR\n\n${content}`, {padding: 1, borderColor: 'red', borderStyle: 'double'});
    return `${prefix}\n${text}`;
};

const formatDebug = (prefix: string, content: string) => {
    const columns = getColumns();
    const wrappedContent = wrap(`${prefix} ${content}`, columns);
    return indentAllExceptFirstLine(wrappedContent);
};

const formatHelp = (prefix: string, content: string) => {
    const text = indentedBox(content, {padding: 1, borderColor: 'blue', borderStyle: 'single'});
    return `${prefix}\n${text}`;
};

export const decorateWithLevel = (level: Level, content: string) => {
    const prefix = `${getDuration()}${getPackageAndFilename(level)}`;
    switch (level) {
        case 'INFO':
            return formatInfo(prefix, content);
        case 'WARN':
            return formatWarn(prefix, content);
        case 'ERROR':
            return formatError(prefix, content);
        case 'DEBUG':
            return formatDebug(prefix, content);
        case 'HELP':
            return formatHelp(prefix, content);
        default:
            return formatInfo(prefix, content);
    }
};
