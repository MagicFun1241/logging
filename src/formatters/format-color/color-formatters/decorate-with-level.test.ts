import {decorateWithLevel} from './decorate-with-level';

jest.mock('../../../config', () => ({
    getColumns: () => 80,
    getColumnsWithBoxen: () => 90,
    logConfig: {indent: 8}
}));
jest.mock('../prefix', () => ({
    getPackageAndFilename: () => 'mock-package/mock-filename',
    getDuration: () => 'mock-duration'
}));

describe('decorate-with-level', () => {
    test('info', async () => {
        const result = decorateWithLevel('INFO', 'content1234567890'.repeat(10));
        expect(result).toMatchSnapshot();
    });

    test('warn', async () => {
        const result = decorateWithLevel('WARN', 'content1234567890'.repeat(10));
        expect(result).toMatchSnapshot();
    });

    test('error', async () => {
        const result = decorateWithLevel('ERROR', 'content1234567890'.repeat(10));
        expect(result).toMatchSnapshot();
    });

    test('debug', async () => {
        const result = decorateWithLevel('DEBUG', 'content1234567890'.repeat(10));
        expect(result).toMatchSnapshot();
    });

    test('help', async () => {
        const result = decorateWithLevel('HELP', 'content1234567890'.repeat(10));
        expect(result).toMatchSnapshot();
    });
});
