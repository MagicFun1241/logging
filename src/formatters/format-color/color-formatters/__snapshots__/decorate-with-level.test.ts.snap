// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decorate-with-level debug 1`] = `
"mock-durationmock-package/mock-filename 
        content1234567890content1234567890content1234567890content1234567890content12345
        67890content1234567890content1234567890content1234567890content1234567890content
        1234567890"
`;

exports[`decorate-with-level error 1`] = `
"mock-durationmock-package/mock-filename
        [31m╔════════════════════════════════════════════════════════════════════════════════════════════════╗[39m
        [31m║[39m                                                                                                [31m║[39m
        [31m║[39m   [37m[41m> ERROR <[49m[39m                                                                                    [31m║[39m
        [31m║[39m                                                                                                [31m║[39m
        [31m║[39m   content1234567890content1234567890content1234567890content1234567890content1234567890conte   [31m║[39m
        [31m║[39m   nt1234567890content1234567890content1234567890content1234567890content1234567890             [31m║[39m
        [31m║[39m                                                                                                [31m║[39m
        [31m╚════════════════════════════════════════════════════════════════════════════════════════════════╝[39m"
`;

exports[`decorate-with-level help 1`] = `
"mock-durationmock-package/mock-filename
        [34m┌────────────────────────────────────────────────────────────────────────────────────────────────┐[39m
        [34m│[39m                                                                                                [34m│[39m
        [34m│[39m   content1234567890content1234567890content1234567890content1234567890content1234567890conte   [34m│[39m
        [34m│[39m   nt1234567890content1234567890content1234567890content1234567890content1234567890             [34m│[39m
        [34m│[39m                                                                                                [34m│[39m
        [34m└────────────────────────────────────────────────────────────────────────────────────────────────┘[39m"
`;

exports[`decorate-with-level info 1`] = `
"mock-durationmock-package/mock-filename content1234567890content1234567890content1234567890co
        ntent1234567890content1234567890content1234567890content1234567890content1234567
        890content1234567890content1234567890"
`;

exports[`decorate-with-level warn 1`] = `
"mock-durationmock-package/mock-filename
        [33m┌────────────────────────────────────────────────────────────────────────────────────────────────┐[39m
        [33m│[39m                                                                                                [33m│[39m
        [33m│[39m   [30m[43m> WARNING <[49m[39m                                                                                  [33m│[39m
        [33m│[39m                                                                                                [33m│[39m
        [33m│[39m   content1234567890content1234567890content1234567890content1234567890content1234567890conte   [33m│[39m
        [33m│[39m   nt1234567890content1234567890content1234567890content1234567890content1234567890             [33m│[39m
        [33m│[39m                                                                                                [33m│[39m
        [33m└────────────────────────────────────────────────────────────────────────────────────────────────┘[39m"
`;
